src/Analyzers/CSharp/Tests/UseIndexOrRangeOperator/UseRangeOperatorTests.cs

```diff
@@ -479,5 +478,64 @@ void Goo(Span<int> s)
                 FixedCode = fixedSource,
             }.RunAsync();
         }
+
+        [WorkItem(43202, "https://github.com/dotnet/roslyn/issues/43202")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseRangeOperator)]
+        public async Task TestWritableType()
+        {
+            var source =
+@"
+using System;
+struct S { 
+    public ref S Slice(int start, int length) => throw null; 
+    public int Length { get; } 
+    public S this[System.Range r] { get => default; } 
```

Comment:
💡 Also add a test like this one, where the property is a read/write property (as opposed to a ref-returning property):

```
public S this[System.Range r] { get => default; set { } }
```